.github/workflows/main.yml

```diff
@@ -33,25 +52,33 @@ jobs:
         restore-keys: |
           ${{ runner.os }}-pip-
 
-    - name: Upgrade packaging tools
-      run: python -m pip install --upgrade pip setuptools virtualenv wheel
-
     - name: Install dependencies
-      run: python -m pip install --upgrade codecov tox
-
-    - name: Run tox targets for ${{ matrix.python-version }}
       run: |
-        ENV_PREFIX=$(tr -C -d "0-9" <<< "${{ matrix.python-version }}")
-        TOXENV=$(tox --listenvs | grep "^py$ENV_PREFIX" | tr '\n' ',') tox
+        python -m pip install --upgrade pip setuptools virtualenv wheel
+        python -m pip install --upgrade codecov tox tox-gh-actions
 
-    - name: Run extra tox targets
-      if: ${{ matrix.python-version == '3.9' }}
+    - name: Run tox
       run: |
-        python setup.py bdist_wheel
-        rm -r djangorestframework.egg-info  # see #6139
-        tox -e base,dist,docs
-        tox -e dist --installpkg ./dist/djangorestframework-*.whl
+        TOX_EXIT=0
+        tox || TOX_EXIT=$?
+
+        if [[ $DJANGO == '3.2' && ${{ matrix.python-version }} == '3.9' ]]; then
```

Comment:
This could be done as before, but I believe it would not be run if the first fails. This should allow both to run and report an error if either fails.